skorecard/skorecard.py

```diff
@@ -156,114 +148,83 @@ def __repr__(self):
                 v = str(v.__class__.__name__)  # get only the class name of the bucketer
             vars += f"{k}={v}, "
 
-        self.repr_msg = f"{self.__class__.__name__}({vars})"
-        return self.repr_msg
+        return f"{self.__class__.__name__}({vars})"
 
-    def _build_default_bucketing_process(self):
-        """Make the default bucketing step of Skorecard as a BucketingProcess."""
-        prebucketing_pipe = [DecisionTreeBucketer(variables=self.num_features, max_n_bins=50, min_bin_size=0.02)]
+    def _build_default_bucketing_process(self, X):
+        """
+        A default BucketingProcess to use if not specified by user.
+        """
+        # Auto-detect cat and numerical columns
+        num_features = list(X._get_numeric_data().columns)
+        cat_features = [f for f in X.columns if f not in num_features]
 
-        bucketing_pipe = [OptimalBucketer(variables=self.num_features, max_n_bins=6, min_bin_size=0.05)]
+        prebucketing_pipe = []
+        bucketing_pipe = []
 
-        # Add this part only if cat_features are defined
-        if self.cat_features:
-            prebucketing_pipe.append(OrdinalCategoricalBucketer(variables=self.cat_features, tol=0.02))
+        if len(num_features) > 0:
+            prebucketing_pipe.append(DecisionTreeBucketer(variables=num_features, max_n_bins=100, min_bin_size=0.02))
+            bucketing_pipe.append(OptimalBucketer(variables=num_features, max_n_bins=6, min_bin_size=0.05))
+        if len(cat_features) > 0:
             bucketing_pipe.append(
                 OptimalBucketer(
-                    variables=self.cat_features,
+                    variables=cat_features,
                     variables_type="categorical",
                     max_n_bins=6,
                     min_bin_size=0.05,
                 )
             )
+        if len(num_features) == 0 and len(num_features) == 0:
```

Comment:
cat_features, you have num_features twice

Reply:
nice catch, fixed